scene/resources/sky_material.cpp

```diff
@@ -322,15 +333,21 @@ RID PanoramaSkyMaterial::get_rid() const {
 }
 
 RID PanoramaSkyMaterial::get_shader_rid() const {
-	_update_shader();
+	MutexLock lock(shader_mutex);
+	((PanoramaSkyMaterial *)this)->_update_shader();
```

Comment:
Mutex locks should have a corresponding unlock (see update shader below)

As above, I am not sure this cast is needed